core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -154,12 +239,130 @@ oppia.directive('questionPlayer', [
+              // Calculate total score
+              ctrl.totalScore += questionScore;
+
+              // Calculate scores per skill
+              if (!(questionData.linkedSkillIds)) {
+                continue;
+              }
+              for (var i = 0; i < questionData.linkedSkillIds.length; i++) {
+                var skillId = questionData.linkedSkillIds[i];
+                if (!(skillId in ctrl.scorePerSkillMapping)) {
+                  continue;
+                }
+                ctrl.scorePerSkillMapping[skillId].score += questionScore;
+                ctrl.scorePerSkillMapping[skillId].total += 1.0;
+              }
+            }
+            ctrl.totalScore = Math.round(
+              ctrl.totalScore * 100 / totalQuestions);
+            $scope.resultsLoaded = true;
+          };
+
+          var passOrFailTests = function() {
+            ctrl.passTest = true;
+            if (isInPassOrFailMode()) {
+              Object.keys(ctrl.scorePerSkillMapping).forEach(function(skillId) {
+                var correctionRate = ctrl.scorePerSkillMapping[skillId].score /
+                  ctrl.scorePerSkillMapping[skillId].total;
+                if (correctionRate < CORRECTION_RATE_TO_PASS) {
+                  ctrl.passTest = false;
+                }
+              });
+            }
+
+            if (!ctrl.passTest) {
+              ctrl.questionPlayerConfig = {};
+            }
+          };
+
+          ctrl.getScorePercentage = function(scorePerSkill) {
+            return scorePerSkill.score / scorePerSkill.total * 100;
+          };
+
+          ctrl.getColorForScore = function(scorePerSkill) {
+            var correctionRate = scorePerSkill.score / scorePerSkill.total;
+            if (correctionRate >= CORRECTION_RATE_TO_PASS) {
+              return 'rgb(0, 150, 136)';
+            } else {
+              return 'rgb(217, 92, 12)';
+            }
+          };
+
           $rootScope.$on('currentQuestionChanged', function(event, result) {
             updateCurrentQuestion(result + 1);
           });
+
           $rootScope.$on('totalQuestionsReceived', function(event, result) {
             updateTotalQuestions(result);
           });
+
+          $rootScope.$on('questionSessionCompleted', function(event, result) {
+            $location.hash(HASH_PARAM +
+              encodeURIComponent(JSON.stringify(result)));
+          });
+
+          $scope.$on('$locationChangeSuccess', function(event) {
+            var hashContent = $location.hash();
+            if (!hashContent || hashContent.indexOf(HASH_PARAM) === -1) {
+              return;
+            }
+            var resultHashString = decodeURIComponent(
+              hashContent.substring(hashContent.indexOf(
+                HASH_PARAM) + HASH_PARAM.length));
+            if (resultHashString) {
+              var questionStateData = JSON.parse(resultHashString);
+              calculateScores(questionStateData);
+              passOrFailTests();
```

Comment:
Change to `$ctrl.isTestPassed = hasUserPassedTest();`

Reply:
Done.